tests/test_lie_operations.py

```diff
@@ -84,6 +84,22 @@ class TestGroupSpecificOperations(absltest.TestCase):
 
     # SO3.
 
+    def test_so3_equality(self):
+        rot_1 = SO3.identity()
+        rot_2 = SO3.identity()
+        self.assertEqual(rot_1, rot_2)
+
+        rot_1 = SO3.from_x_radians(np.pi)
+        rot_2 = SO3.from_x_radians(np.pi)
+        self.assertEqual(rot_1, rot_2)
+
+        rot_1 = SO3.from_x_radians(np.pi)
+        rot_2 = SO3.from_x_radians(np.pi * 0.5)
+        self.assertNotEqual(rot_1, rot_2)
+
+        # Make sure different types are properly handled
+        self.assertNotEqual(SO3.identity(), 5)
```

Comment:
I'm surprised you didn't need a `with self.assertRaises` catch here. Should it not be throwing a NotImplemented error?

Reply:
`NotImplemented` in the `__eq__` context will signify to the interpreter that the reflexive equality check should be attempted since `self` does not know how to compare itself to `other` (but maybe `other` knows how to compare itself to `self`). If the reflexive operation also returns `NotImplemented`, then `__eq__` returns `False`. This is a little different from `NotImplementedError` - here's a [reference](https://docs.python.org/3/library/constants.html#NotImplemented) 🙂

Reply:
TIL!